### ISSUE TITLE
Tests don't complete under multi-archive setup (v3.3.10)

### ISSUE THREAD

**Issue by jesusbagpuss** [eprintsug/hefce_oa#5]
The test harness in EPrints returns the first (alphabetically?) defined archive.
https://github.com/eprints/eprints/blob/3.3/perl_lib/EPrints/Test.pm#L35-L45

This may well _not_ be the hefce_oa enabled repository.

**Comment by jesusbagpuss** [comment]
Changing e.g. this https://github.com/eprintsug/hefce_oa/blob/master/tests/ACC_EMBARGO.t#L10 to the code below seems to work - but I'm not sure if it's the most efficient/correct way?

``` perl
use Cwd 'abs_path';
my $repoid = abs_path(__FILE__);
$repoid =~ s/^.*?\/archives\/([^\/]+)\/tests\/.*?$/$1/;
my $repo = EPrints::Repository->new( $repoid );
```

Ideally you would you be able to pass a repoid to EPrints::Test.

**Comment by drtjmb** [comment]
See my latest comment on https://github.com/eprintsug/hefce_oa/pull/8 - if this works for you please create a pull request on eprints/eprints

**Comment by jesusbagpuss** [comment]
Fixed by #8
